Assets/Source/Components/NavigationMesh/NavigationMeshComponent.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Assets.Source.AStar;
+using Assets.Source.Components.Base;
+using System.Linq;
+using UnityEngine;
+
+namespace Assets.Source.Components.NavigationMesh
+{
+    public class NavigationMeshComponent : ComponentBase
+    {
+        [Tooltip("How many tiles going horizontally")]
+        [SerializeField]
+        private int gridWidth = 20;
+        public int GridWidth => gridWidth;
+
+        [Tooltip("How many tiles going vertically")]
+        [SerializeField]
+        private int gridHeight = 15;
+        public int GridHeight => gridHeight;
+
+
+        [Tooltip("The size of each tile in units.  Should be large enough for a pathfinder to pass through " +
+            "without clipping the edge of corners.")]
+        [SerializeField]
+        private float tileSize = 0.5f;
+
+        [Tooltip("Recalculates the entire navigation mesh every frame.  Leave this on if things in your " +
+            "level move around, or set to false if the level is static for a performance boost.")]
+        [SerializeField]
+        private bool syncEveryFrame = true;
+
+        private Node[][] nodes;
+
+        public override void ComponentAwake()
+        {
+            nodes = new Node[gridWidth][];
+            GenerateNavigationMesh();
+            base.ComponentAwake();
+        }
+                
+        // Generate the initial nav mesh
+        private void GenerateNavigationMesh()
+        {
+            for (var ix = 0; ix < gridWidth; ix++)
+            {
+                nodes[ix] = new Node[gridHeight];
+                for (var iy = 0; iy < gridHeight; iy++)
+                {
+                    var center = new Vector2(x: (ix * tileSize) + transform.position.x, y: (iy * tileSize) + transform.position.y);
+                    var isSolid = TileContainsSolid(center);
+
+                    nodes[ix][iy] = new Node(center, ix, iy, isSolid);
+                }
+            }
+        }
+
+        // Returns true if any solid exists in a radius of 'tileSize' around the specified center position
+        private bool TileContainsSolid(Vector2 center) => Physics2D.OverlapArea(new Vector2(center.x - (tileSize / 2), center.y - (tileSize / 2)),
```

Comment:
I think it might just be github code review but this is really hard to look at, could you use a lambda with a statement block instead to make this look more like method? Then you can split up the calculating of x and y for this vector2

Reply:
I could try it, but im not super sure that this code will ever really be all that easy to read lol.